cmake/cyclonedds/CMakeLists.txt

```diff
@@ -70,7 +70,13 @@ function(fetch_and_install name)
     endforeach()
     string( REPLACE ";" " " ADDITIONAL_CMAKE_FLAGS "${ADDITIONAL_CMAKE_FLAGS}")
 
-    execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${ADDITIONAL_CMAKE_FLAGS}" "${TOOLCHAIN_FILE}" "${EXTRA_CMAKE_ARGS}"  "${SOURCE_DIR}"
+    set(CYCLONE_BUILD_ARGS "-DCMAKE_INSTALL_PREFIX=${INSTALL_DIR}" "${EXTRA_CMAKE_ARGS}" "${ADDITIONAL_CMAKE_FLAGS}" "${SOURCE_DIR}")
+
+    if(NOT ADDITIONAL_CMAKE_FLAGS STREQUAL "")
+        list(APPEND CYCLONE_BUILD_ARGS ${ADDITIONAL_CMAKE_FLAGS})
+    endif()
```

Comment:
`ADDITIONAL_CMAKE_FLAGS` are added twice to `CYCLONE_BUILD_ARGS` ... your really want to be sure, right ;)